src/apiClient.ts

```diff
@@ -156,7 +156,25 @@ export default class APIClient {
   private async sendRequest(options: RequestOptionsParams): Promise<Response> {
     const req = this.newRequest(options);
     const controller: AbortController = new AbortController();
-    const timeoutDuration = options.overrides?.timeout || this.timeout;
+    
+    // Handle timeout backward compatibility
+    let timeoutDuration: number;
+    if (options.overrides?.timeout) {
+      // Determine if the override timeout is likely in milliseconds (≥ 1000)
+      if (options.overrides.timeout >= 1000) {
+        // eslint-disable-next-line no-console
+        console.warn(
+          'DEPRECATED: Providing timeout in milliseconds via request overrides is deprecated and will be removed in the next major release. Please use seconds instead.'
+        );
```

Comment:
@ajay-k let's remove this console warn message and instead add a `@deprecated [message here]` to the `options.overrides.timeout` property defined in the `OverridableNylasConfig` interface/type. So we'll just use typescript to notify of this impending deprecation.

Also update the CHANGELOG.md to call out this deprecation.

Finally, you have some github actions failing that need to be resolved.

Thank you!

Reply:
Hey @AaronDDM made some changes you suggested, can you review and lmk if it looks good to you as well?